src/test/java/uk/gov/companieshouse/api/accounts/service/impl/CreditorsWithinOneYearServiceTest.java

```diff
@@ -0,0 +1,127 @@
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.dao.DuplicateKeyException;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.links.BasicLinkType;
+import uk.gov.companieshouse.api.accounts.model.entity.notes.CreditorsWithinOneYearDataEntity;
+import uk.gov.companieshouse.api.accounts.model.entity.notes.CreditorsWithinOneYearEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.CreditorsWithinOneYear;
+import uk.gov.companieshouse.api.accounts.repository.CreditorsWithinOneYearRepository;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseStatus;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transformer.CreditorsWithinOneYearTransformer;
+import uk.gov.companieshouse.api.accounts.utility.impl.KeyIdGenerator;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class CreditorsWithinOneYearServiceTest {
+
+    @InjectMocks
+    private CreditorsWithinOneYearService service;
+
+    @Mock
+    private CreditorsWithinOneYearTransformer mockTransformer;
+
+    @Mock
+    private CreditorsWithinOneYear mockCreditorsWithinOneYear;
+
+    @Mock
+    private Transaction mockTransaction;
+
+    @Mock
+    private HttpServletRequest mockRequest;
+
+    @Mock
+    private CreditorsWithinOneYearRepository mockRepository;
+
+    @Mock
+    private DuplicateKeyException mockDuplicateKeyException;
+
+    @Mock
+    private MongoException mockMongoException;
+
+    @Mock
+    private SmallFullService mockSmallFullService;
+
+    @Mock
+    private KeyIdGenerator mockKeyIdGenerator;
+
+    private CreditorsWithinOneYearEntity creditorsWithinOneYearEntity;
+
+
+    @BeforeEach
+    void setUp() {
+
+        CreditorsWithinOneYearDataEntity dataEntity = new CreditorsWithinOneYearDataEntity();
+
+        Map<String, String> links = new HashMap<>();
+        links.put(BasicLinkType.SELF.getLink(), "self_link");
+        dataEntity.setLinks(links);
+
+        creditorsWithinOneYearEntity = new CreditorsWithinOneYearEntity();
+        creditorsWithinOneYearEntity.setData(dataEntity);
+    }
+
+    @Test
+    @DisplayName("Tests the successful creation of a Creditors within one year resource")
+    void canCreateCreditorsWithinOneYear() throws DataException {
```

Comment:
You could assert in this test that metadata (etag, links and kind) are set on the rest object to ensure the created resource has all appropriate data. I'd also assert that the small full service is called to add the link for the creditors within one year resource.

Reply:
done